pymeasure/instruments/santec/tsl570.py

```diff
@@ -0,0 +1,263 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument, SCPIMixin
+from pymeasure.instruments.validators import (
+    strict_range,
+    strict_discrete_range,
+    strict_discrete_set,
+)
+from pymeasure.errors import Error
+
+
+class TSL570(SCPIMixin, Instrument):
+    """Represents the Santec TSL-570 Tunable Laser and provides a high-level interface for
+    interacting with the instrument."""
+
+    def __init__(self, adapter, name="Santec TSL-570", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        self.write(":SYSTem:COMMunicate:CODe 1")  # Set the device to use SCPI commands
+        self.check_errors()  # Clear the error queue
+
+    def check_instr_errors(self):
```

Comment:
I think this method is superfluous, as the parameter `check_get_errors` does, what you want to do here.